tests/rbd/test_rbd_encryption.py

```diff
@@ -0,0 +1,314 @@
+        return 0
+    except Exception as error:
+        log.error(str(error))
+        return 1
+    finally:
+        log.info("Clean up parent device")
+        encrypt_config = list()
+        if parent_encryption_type.startswith("luks"):
+            encrypt_config.append({parent_encryption_type: passphrase})
+        device_cleanup(
+            rbd=rbd,
+            file_name=parent_file_path.rsplit("/", 1)[0],
+            image_spec=image_spec,
+            encryption_config=encrypt_config,
+        )
+
+        log.info("Cleanup clone device")
+        if clone_encryption_type.startswith("luks"):
+            encrypt_config = encryption_config
+        device_cleanup(
+            rbd=rbd,
+            file_name=clone_file_path.rsplit("/", 1)[0],
+            image_spec=clone_spec,
+            encryption_config=encrypt_config,
+        )
+
+
+def cleanup(rbd_obj, **kw):
+    """Cleanup the replicated and ecpools created for the test module
+
+    Args:
+        rbd_obj (dict): dictionary containing rbd ecpool and replicated pool object
+    """
+    log.info("Cleaning up pools")
+    obj = rbd_obj.get("rbd_ecpool", rbd_obj["rbd_reppool"])
+    obj.clean_up(
+        pools=[
+            kw["config"].get("rep_pool_config", {})["pool"],
+            kw["config"].get("ec_pool_config", {})["pool"],
+        ]
+    )
+
+
+def run(**kw):
+    """Verify the luks encryption functionality for image and clone
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common, fio and rbd-nbd packages,
+    conf and keyring files
+
+    Test cases covered -
+    1) CEPH-83575263 - Encrypt & decrypt file using same keys and different keys
+
+    kw:
+        config:
+            encryption_format: #parent,clone
+                - luks1, luks1
+    Test Case Flow
+    1. Create a pool and an Image
+    2. Apply Encryption LUKS1 format to the image
+    3. Perform resize operations to compensate the overhead due to LUKS1 header
+    4. Load encryption using rbd device map
+    5. Mount the image and run IOs
+    6. Create snapshot, Protect the snapshot and run IOs
+    7. Perform clone from the snapshot of the image created
+    8. Apply encryption with different passphrase key other than parent image
+    9. Load the encryption using rbd device map on cloned image
+    10. Mount the clone and run Ios
+    11. Flatten the clone
+    12. Verify data integrity between clone image and the original image
+
+    2) CEPH-83575251 - Apply different combinations of luks encryption on an RBD image
+    and its clone and verify data integrity between parent and clone
+
+    kw:
+        config:
+            encryption_format: #parent,clone
+                - NA, luks1
+                - NA, luks2
+                - luks1, luks1
+                - luks1, luks2
+                - luks1, NA
+                - luks2, luks1
+                - luks2, luks2
+                - luks2, NA
+    Test Case Flow
+    1. Create a pool and an Image
```

Comment:
Does this test need to cover on both EC pool as well?

Reply:
Yes it is covering both replicated pool and ec pool